vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java

```diff
@@ -58,4 +60,19 @@ void shouldInvokeRunnableOnceBeforeReachingStopCondition() throws IOException
         new DurationBasedWaiter(Duration.ZERO, Duration.ZERO).wait(failableRunnable, Boolean.TRUE::booleanValue);
         verify(failableRunnable, times(1)).run();
     }
+
+    @Test
+    void shouldDecreasePollingTimeoutAccordinglyAfterLongRunningRunnable() throws IOException
+    {
+        FailableSupplier<Boolean, IOException> valueProvider = mock(FailableSupplier.class);
+        when(valueProvider.get())
+                .thenAnswer(invocation -> {
+                    Sleeper.sleep(Duration.ofSeconds(5));
```

Comment:
Why is the timeout so big, any particular reason for this?

Reply:
runnable execution time is 5 times bigger than polling timeout, having such values we can guarantee no race condition will be met (especially on slow machines)

Reply:
another option is to mock `sleep`-s

Reply:
https://github.com/vividus-framework/vividus/search?q=slow&type=commits - in general to avoid such "fine"-tunnings